src/camera.rs

```diff
@@ -128,48 +142,31 @@ const OPENGL_TO_WGPU_MATRIX: Matrix4<f32> = Matrix4::new(
     0.0, 0.0, 0.5, 1.0,
 );
 
-#[derive(Copy, Clone, Debug, Default)]
+#[derive(Clone, Debug, Default)]
 pub enum Target {
     Origin,
     #[default]
     FabricMidpoint,
-    SelectedFace(UniqueId),
+    AroundFaces(HashSet<UniqueId>),
 }
 
 impl Target {
     pub fn look_at(&self, fabric: &Fabric) -> Option<Point3<f32>> {
         match self {
             Target::Origin => Some(point3(0.0, 0.0, 0.0)),
             Target::FabricMidpoint => Some(fabric.midpoint()),
-            Target::SelectedFace(face_id) => {
-                fabric.faces.get(face_id).map(|face| {
-                    Point3::from_vec(face.midpoint(fabric))
-                })
+            Target::AroundFaces(face_set) => {
+                let midpoints = face_set
+                    .iter()
+                    .flat_map(|face_id|
```

Comment:
```suggestion
                    .filter_map(|face_id|
```
flat_map() technically works the same, but should only be used to flatten iterables. Here you're just filtermapping Options